Mathlib/Geometry/Manifold/Instances/Sphere.lean

```diff
@@ -575,3 +577,87 @@ theorem contMDiff_circleExp : ContMDiff 𝓘(ℝ, ℝ) (𝓡 1) ∞ Circle.exp :
 @[deprecated (since := "2024-07-25")] alias contMDiff_expMapCircle := contMDiff_circleExp
 
 end Circle
+
+section OnePoint
+
+/-
+# One-point compactification of Euclidean space is homeomorphic to the sphere.
+-/
+
+variable {n : ℕ}
+variable {v : EuclideanSpace ℝ (Fin n.succ)}
+variable (hv : ‖v‖ = 1)
+variable (hv' : v ∈ sphere 0 (1:ℝ))
+/-
+    For example, `v` could be
+    `(EuclideanSpace.single (0:Fin n.succ) (1:ℝ))`
+-/
+
+open Set Submodule
+
+/-- The inverse stereographic projection is injective. -/
+theorem stereo_inj : Injective (stereoInvFun hv) := fun x y h =>
+  (stereographic hv).right_inv' (show x ∈ (stereographic hv).target by trivial) ▸
+  (stereographic hv).right_inv' (show y ∈ (stereographic hv).target by trivial) ▸
+  congrArg (stereographic hv).toPartialEquiv h
+
+
+/-- Instance needed by `Submodule_homeo_Euclidean`. -/
+instance {n : ℕ} : Fact (finrank ℝ (EuclideanSpace ℝ (Fin n.succ)) = n + 1) := {out := by simp}
+
+/-- A codimension 1 subspace of Euclidean space is homeomorphic to a sphere. -/
+noncomputable def Submodule_homeo_Euclidean {n : ℕ}
+    (v : sphere (0 : (EuclideanSpace ℝ (Fin n.succ))) 1) :
+    Homeomorph ((span ℝ {v.1})ᗮ) (EuclideanSpace ℝ (Fin n)) :=
+  (OrthonormalBasis.fromOrthogonalSpanSingleton n (ne_zero_of_mem_unit_sphere v)).repr.toHomeomorph
```

Comment:
```suggestion
/-- The orthogonal complement of the span of a point on the sphere
is homeomorphic to a Euclidean space of codimension 1. -/
noncomputable def Submodule_homeo_Euclidean {n : ℕ}
    (v : sphere (0 : (EuclideanSpace ℝ (Fin n.succ))) 1) :
    Homeomorph ((span ℝ {v.1})ᗮ) (EuclideanSpace ℝ (Fin n)) :=
  letI fact {n : ℕ} : Fact (finrank ℝ (EuclideanSpace ℝ (Fin n.succ)) = n + 1) := ⟨by simp⟩
  (OrthonormalBasis.fromOrthogonalSpanSingleton n (ne_zero_of_mem_unit_sphere v)).repr.toHomeomorph
```

- We try to have as little global instances of `Fact` as possible. Hence I'm inlining it.
- I don't think the original docstring is what you intended.

Reply:
Thanks. And sorry about the docstring.